lib/cell.rb

```diff
@@ -0,0 +1,58 @@
+require './lib/coordinate_validation'
+
+class Cell
+  attr_reader :coordinate,
+              :ship
+
+  include CoordinateValidation
+
+  # Initialize a cell with a coordinate and no ship to start
+  def initialize(coordinate)
+    validate_coordinate(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @fired_upon = false
+  end
+
+  # Check if a cell is empty
+  def empty?
+    @ship.nil?
+  end
+
+  # Place a ship in a cell
+  def place_ship(ship)
+    @ship = ship
+  end
+
+  # Fire upon a cell
+  def fire_upon
+    # The @ship&.hit means: the method call hit should only be performed if @ship is not null
+    @ship&.hit
+    @fired_upon = true
+  end
+
+  # Check if a cell has been fired upon
+  def fired_upon?
+    @fired_upon
+  end
+
+  # returns a String representation of the Cell for when we need to print the board
+  #   ”.” if the cell has not been fired upon.
+  # “M” if
+  # “H” if
+  # “X” if the cell has been fired upon and its ship has been sunk.
+
+  def render(reveal_ship = false)
+    # The cell has been fired upon and it does not contain a ship (the shot was a miss).
+    return 'M' if fired_upon? == true && empty?
```

Comment:
For boolean methods, no need to include `== true`, you could jsut say `if fired_upon?` or `if !fired_upon?`